### ISSUE TITLE
Error thrown when running resave/gift-voucher-codes console command

### ISSUE THREAD

**Issue by jmauzyk** [verbb/gift-voucher#83]
**Description**
When running `./craft resave/gift-voucher-codes` console command I get the error below. I believe this is specific to PostgreSQL environments, as I tested in MySQL and was able to successfully run the command.

```
Exception 'yii\db\Exception' with message 'SQLSTATE[42803]: Grouping error: 7 ERROR:  column "giftvoucher_codes.dateCreated" must appear in the GROUP BY clause or be used in an aggregate function
LINE 14: ORDER BY "giftvoucher_codes"."dateCreated" DESC
                  ^
The SQL being executed was: SELECT COUNT(*)
FROM (SELECT "elements"."id" AS "elementsId", "elements_sites"."id" AS "elementsSitesId", "content"."id" AS "contentId"
FROM "craft_elements" "elements"
INNER JOIN "craft_giftvoucher_codes" "giftvoucher_codes" ON "giftvoucher_codes"."id" = "elements"."id"
INNER JOIN "craft_giftvoucher_vouchers" "giftvoucher_vouchers" ON "giftvoucher_codes"."voucherId" = "giftvoucher_vouchers"."id"
INNER JOIN "craft_elements_sites" "elements_sites" ON "elements_sites"."elementId" = "elements"."id"
INNER JOIN "craft_content" "content" ON "content"."elementId" = "elements"."id"
WHERE ("elements"."archived"=FALSE) AND ("elements"."dateDeleted" IS NULL) AND ("elements"."draftId" IS NULL) AND ("elements"."revisionId" IS NULL)
ORDER BY "giftvoucher_codes"."dateCreated" DESC) "subquery"
INNER JOIN "craft_giftvoucher_codes" "giftvoucher_codes" ON "giftvoucher_codes"."id" = "subquery"."elementsId"
INNER JOIN "craft_elements" "elements" ON "elements"."id" = "subquery"."elementsId"
INNER JOIN "craft_elements_sites" "elements_sites" ON "elements_sites"."id" = "subquery"."elementsSitesId"
INNER JOIN "craft_content" "content" ON "content"."id" = "subquery"."contentId"
ORDER BY "giftvoucher_codes"."dateCreated" DESC'

in /app/vendor/yiisoft/yii2/db/Schema.php:676

Error Info:
Array
(
    [0] => 42803
    [1] => 7
    [2] => ERROR:  column "giftvoucher_codes.dateCreated" must appear in the GROUP BY clause or be used in an aggregate function
LINE 14: ORDER BY "giftvoucher_codes"."dateCreated" DESC
                  ^
)

Caused by: Exception 'PDOException' with message 'SQLSTATE[42803]: Grouping error: 7 ERROR:  column "giftvoucher_codes.dateCreated" must appear in the GROUP BY clause or be used in an aggregate function
LINE 14: ORDER BY "giftvoucher_codes"."dateCreated" DESC
                  ^'

in /app/vendor/yiisoft/yii2/db/Command.php:1302

Stack trace:
#0 /app/vendor/yiisoft/yii2/db/Command.php(1302): PDOStatement->execute()
#1 /app/vendor/yiisoft/yii2/db/Command.php(1168): yii\db\Command->internalExecute('SELECT COUNT(*)...')
#2 /app/vendor/yiisoft/yii2/db/Command.php(436): yii\db\Command->queryInternal('fetchColumn', 0)
#3 /app/vendor/yiisoft/yii2/db/Query.php(497): yii\db\Command->queryScalar()
#4 /app/vendor/craftcms/cms/src/db/Query.php(258): yii\db\Query->queryScalar('COUNT(*)', NULL)
#5 /app/vendor/yiisoft/yii2/db/Query.php(368): craft\db\Query->queryScalar('COUNT(*)', NULL)
#6 /app/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1604): yii\db\Query->count('*', NULL)
#7 /app/vendor/craftcms/cms/src/console/controllers/ResaveController.php(397): craft\elements\db\ElementQuery->count()
#8 /app/vendor/craftcms/cms/src/console/controllers/ResaveController.php(341): craft\console\controllers\ResaveController->_resaveElements(Object(verbb\giftvoucher\elements\db\CodeQuery))
#9 /app/vendor/verbb/gift-voucher/src/GiftVoucher.php(381): craft\console\controllers\ResaveController->saveElements(Object(verbb\giftvoucher\elements\db\CodeQuery))
#10 [internal function]: verbb\giftvoucher\GiftVoucher->verbb\giftvoucher\{closure}()
#11 /app/vendor/craftcms/cms/src/console/CallableAction.php(41): call_user_func_array(Object(Closure), Array)
#12 /app/vendor/yiisoft/yii2/base/Controller.php(178): craft\console\CallableAction->runWithParams(Array)
#13 /app/vendor/yiisoft/yii2/console/Controller.php(182): yii\base\Controller->runAction('gift-voucher-co...', Array)
#14 /app/vendor/craftcms/cms/src/console/Controller.php(221): yii\console\Controller->runAction('gift-voucher-co...', Array)
#15 /app/vendor/yiisoft/yii2/base/Module.php(552): craft\console\Controller->runAction('gift-voucher-co...', Array)
#16 /app/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('resave/gift-vou...', Array)
#17 /app/vendor/craftcms/cms/src/console/Application.php(89): yii\console\Application->runAction('resave/gift-vou...', Array)
#18 /app/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('resave/gift-vou...', Array)
#19 /app/vendor/yiisoft/yii2/base/Application.php(384): yii\console\Application->handleRequest(Object(craft\console\Request))
#20 /app/craft(13): yii\base\Application->run()
#21 {main}
```

**Additional info**
- Plugin version: 2.7.0
- Craft version: Pro 3.7.33
- Multi-site: No

**Comment by engram-design** [comment]
Fixed for the next release. To get the fix early, change your `verbb/gift-voucher` requirement in `composer.json` to:

```
"require": {
  "verbb/gift-voucher": "dev-craft-3 as 2.7.0",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [2.7.1](https://github.com/verbb/gift-voucher/releases/tag/2.7.1)
